lib/ansible/modules/cloud/amazon/route53_zone.py

```diff
@@ -204,11 +204,12 @@ def create_private(conn, matching_zones, vpc_id, vpc_region, zone_in, record):
             changed = False
             return changed, record
 
-    result = conn.create_hosted_zone(zone_in, **record)
-    hosted_zone = result['CreateHostedZoneResponse']['HostedZone']
-    zone_id = hosted_zone['Id'].replace('/hostedzone/', '')
-    record['zone_id'] = zone_id
-    record['name'] = zone_in
+    if not module.check_mode:
```

Comment:
This approach adds a lot of complexity, since we're jumping into every place a write happens and adding a conditional. I'd prefer an approach that split out the "checking for differences" work from the "actually writing changes" work. That way, we'd have fewer new conditionals to manage.